out/WIPL.asm

```diff
@@ -771,55 +747,49 @@ L_055a:
 055c:    c3 42        ld BL, (PC+0x42)
 055e:    49           sub! BL, AL
 055f:    14 02        bz L_0563
-0561:    7b 01        call (PC+0x01) L_0564
+0561:    7b 01        call (PC+0x01) PrintChar
 
 L_0563:
 0563:    09           ret
 
-L_0564:
+PrintChar:
 0564:    7e 81        long_call
-0566:    55 98        mov EX, EX
-0568:    f2 00 f6     st BX, @(0x00f6)
-056b:    19 0d        ble L_057a
-056d:    c0 c5        ld BL, #0xc5
-056f:    f6 39        st BX, unknown
-0571:    00           HALT
+0566:    55 98 f2 00  mov EX, #0xf200	 ; Serisl port 0 base address
+056a:    f6 19 0d     st AL, +0xd(EX)
+056d:    c0 c5        ld BL, #0xc5	 ; Set up baud rate ?
+056f:    f6 39 00     st BL, +0x0(EX)
 0572:    c0 8c        ld BL, #0x8c
-0574:    49           sub! BL, AL
-0575:    14 13        bz L_058a
+0574:    49           sub! BL, AL	 ; If character is 0x0c (form feed)...
+0575:    14 13        bz L_058a	 ; Print with a delay
 0577:    c0 8d        ld BL, #0x8d
-0579:    49           sub! BL, AL
-
-L_057a:
+0579:    49           sub! BL, AL	 ; If character is \r...
 057a:    15 14        bnz L_0590
-057c:    7b 17        call (PC+0x17) L_0595
-057e:    80 8a        ld AL, #0x8a
-0580:    7b 13        call (PC+0x13) L_0595
+057c:    7b 17        call (PC+0x17) RawPrintChar
+057e:    80 8a        ld AL, #0x8a	 ; Add \n
+0580:    7b 13        call (PC+0x13) RawPrintChar
 0582:    80 8d        ld AL, #0x8d
-0584:    a3 1a        st AL, (PC+0x1a)
+0584:    a3 1a        st AL, (PC+0x1a)	 ; (0x5a0) - preserve the original character ???
 0586:    0e           delay 4.5ms
 0587:    7f 81        clear_data_bank??
 0589:    09           ret
 
 L_058a:
-058a:    7b 09        call (PC+0x09) L_0595
+058a:    7b 09        call (PC+0x09) RawPrintChar
 058c:    0e           delay 4.5ms
 058d:    7f 81        clear_data_bank??
 058f:    09           ret
 
 L_0590:
-0590:    7b 03        call (PC+0x03) L_0595
+0590:    7b 03        call (PC+0x03) RawPrintChar
 0592:    7f 81        clear_data_bank??
 0594:    09           ret
 
-L_0595:
-0595:    f6 38        st BX, unknown
-0597:    00           HALT
+RawPrintChar:
+0595:    f6 38 00     ld BL, +0x0(EX)
 0598:    24 31        srl BL, 2
-059a:    11 f9        bnc L_0595
-059c:    f6 19        st BX, unknown
-059e:    01           nop
-059f:    a0 bd        st AL, #0xbd
+059a:    11 f9        bnc RawPrintChar
+059c:    f6 19 01     st AL, +0x1(EX)
+059f:    a0 bd        st AL, #0xbd	 ; Shouldn't this be ld ?
```

Comment:
No. 

The immediate addressing mode actually functions as operating on address PC+1, with post-increment. So it's modifying the second byte of the st AL, #0xnn operand with the current value in AL. Basically an inline variable instruction. 

The `ld BL, (PC+0x42)` instruction at `055c:` then reads this inline variable later. Maybe it's used elsewhere too?

Reply:
Also at `0584:`